lms/static/scripts/frontend_apps/components/FileList.js

```diff
@@ -71,6 +71,24 @@ export default function FileList({
           </Fragment>
         )}
       />
+      {!isLoading && files.length === 0 && (
+        <Fragment>
+          <div className="FileList__no-files">
+            There are no supported files in the current Canvas course.
+          </div>
+          <p>
+            Please see the Canvas help article for{' '}
+            <a
+              href="https://community.canvaslms.com/t5/Instructor-Guide/How-do-I-upload-a-file-to-a-course/ta-p/618"
+              target="_blank"
+              rel="noreferrer"
+            >
+              uploading files to a course
+            </a>{' '}
+            and try again.
+          </p>
```

Comment:
The text looks good. This text is specific to a particular LMS whereas the `FileList` component itself is currently not. One change that we plan to make soon is to re-use this component with other LMSes such as Blackboard.

What I would suggest is that we keep the basic structure here, but pass in the content of the `<div>` and the `<p>` as props from the parent `LMSFilePicker` component. The parent can then pass in the appropriate message text depending on the LMS.

Reply:
I implemented your suggestion.

Your approach is different from [Sean's proposal](https://github.com/hypothesis/lms/pull/2100) which make the FileList less agnostic.